src/dbnode/storage/index/convert/convert.go

```diff
@@ -180,6 +178,18 @@ func FromSeriesIDAndTagIter(id ident.ID, tags ident.TagIterator) (doc.Metadata,
 	return d, nil
 }
 
+func findSliceOrClone(id, tag []byte, expectedIdx int) ([]byte, int) {
+	n := len(tag)
+	expectedEnd := expectedIdx + n
```

Comment:
Nit: worth renaming `expectedIdx` to `expectedStart` to match the `expectedEnd`?